sandbox/apparmor/profile.go

```diff
@@ -251,3 +303,86 @@ var SnapConfineDistroProfilePath = func() string {
 
 	return ""
 }
+
+var UserHomedirs = func() ([]string, error) {
+	sspPath := dirs.SnapSystemParamsUnder(dirs.GlobalRootDir)
+	ssp, err := sysparams.Open(sspPath)
+	if err != nil {
+		return nil, err
+	}
+	return strutil.CommaSeparatedList(ssp.Homedirs), nil
+}
+
+// SetupSnapConfineSnippets inspects the system and sets up local apparmor
+// policy for snap-confine. Local policy is included by the system-wide policy.
+// Returns whether any modifications was made to the snap-confine snippets.
+func SetupSnapConfineSnippets() (wasChanged bool, err error) {
```

Comment:
moving these here should probably be done separately as this PR is rather largish and mixing refactorings and new code is a bit messy